packages/amplify_authenticator/lib/src/widgets/form_fields.dart

```diff
@@ -0,0 +1,344 @@
+import 'package:flutter/material.dart';
+import 'package:amplify_authenticator/src/widgets/containers.dart';
+import 'package:amplify_authenticator/src/state/inherited_auth_viewmodel.dart';
+import 'package:amplify_authenticator/src/utils/validators.dart';
+
+/// Sign In Form Field
+class SignInFormField extends StatelessWidget {
+  /// Requires a custom title, hint text, a type (username, password, etc)
+  /// and an optional callback for input validation.
+  const SignInFormField(
+      {Key? key,
+      required this.title,
+      required this.hintText,
+      required this.type,
+      this.validator})
+      : super(key: key);
+
+  /// Custom title for this form field
+  final String? title;
+
+  /// Custom hint text for this form field
+  final String? hintText;
+
+  /// This form field currently supports the following form field types:
+  /// * Username
+  /// * Email
+  /// * Phone number
+  /// * Password
+  final String? type;
+
+  /// callback meant to validate inputs of this form field.
+  final String? Function(String?)? validator;
+
+  @override
+  Widget build(BuildContext context) {
+    final _authModelView = InheritedAuthViewModel.of(context)!.signInViewModel;
+    var _obscureText = false;
+    var _callBack;
+    Key? _key;
+    String? Function(String?)? _validator;
+    TextInputType? _keyboardType;
+
+    switch (type) {
+      case 'username':
```

Comment:
Consider, if time allows, making these enums. You can use extensions, as well, if the property will be reused in lots of places and you think it looks cleaner (It may not always). If you do, though, extension naming should follow a pattern - I kinda like `MyEnumX` for an enum called `MyEnum`, i.e. `extension MyEnumX on MyEnum`.

Example:
Without extensions:
```dart
enum MyEnum { a, b }

MyEnum e = ...;  
Widget displayWidget;
switch (e) {
  case MyEnum.a:
    displayWidget = Text('a');
    break;
  case MyEnum.b:
    displayWidget = Text('b');
    break;
}
```

With extensions:
```dart
enum MyEnum { a, b }

extension MyEnumX on MyEnum {
  Widget get displayWidget {
    switch (this) {
      case MyEnum.a:
        return Text('a');
      case MyEnum.b:
        return Text('b');
    }
  }
}

MyEnum e = ...;
Widget displayWidget = e.displayWidget;
```